dag/image.go

```diff
@@ -127,11 +132,9 @@ func (img *Image) doRebuild(newTag string, localOnly, disableRunTests bool) erro
 
 	if !disableRunTests {
 		logrus.Infof("Running tests for \"%s:%s\"", img.Name, newTag)
-		return img.runTests(fmt.Sprintf("%s:%s", img.Name, newTag), img.Dockerfile.ContextPath)
-	}
-
-	if err := dockerfile.ResetFromTag(*img.Dockerfile, newTag); err != nil {
-		return fmt.Errorf("failed to reset tag in dockerfile %s: %w", img.Dockerfile.ContextPath, err)
+		if err := img.runTests(fmt.Sprintf("%s:%s", img.Name, newTag), img.Dockerfile.ContextPath); err != nil {
+			return err
+		}
```

Comment:
Tu peux simplifier
```suggestion
		return img.runTests(fmt.Sprintf("%s:%s", img.Name, newTag), img.Dockerfile.ContextPath)
```

Reply:
Non justement, je voulais que ce soit explicite. Sinon si on ajoute un bloc après, il sera jamais executé si on lance les tests. C'était un des problèmes avant

Reply:
Comme tu veux, moi je préfère les early-returns, c'est plus simple à lire. En vrai j'aurais même fait comme ça:

```suggestion
		if disableRunTests {
			return nil
		}

		return img.runTests(fmt.Sprintf("%s:%s", img.Name, newTag), img.Dockerfile.ContextPath)
```

Ça me va si on laisse comme ça je chipote

Reply:
cette version ça me va. le return systématique me dérangeait plus. j'ai updaté